samples/azure/self-host/Core_6/HostWorker/SafeRoleEnvironment.cs

```
@@ -0,0 +1,177 @@
+
+            var instance = roleEnvironmentType.GetProperty("CurrentRoleInstance").GetValue(null, null);
+            return (string)roleInstanceType.GetProperty("Id").GetValue(instance, null);
+        }
+    }
+
+    public static string DeploymentId
+    {
+        get
+        {
+            if (!IsAvailable) throw new Exception("Role environment is not available, please check IsAvailable before calling this property!");
+
+            return (string)roleEnvironmentType.GetProperty("DeploymentId").GetValue(null, null);
+        }
+    }
+    public static string CurrentRoleName
+    {
+        get
+        {
+            if (!IsAvailable) throw new Exception("Role environment is not available, please check IsAvailable before calling this property!");
+
+            var instance = roleEnvironmentType.GetProperty("CurrentRoleInstance").GetValue(null, null);
+            var role = roleInstanceType.GetProperty("Role").GetValue(instance, null);
+            return (string)roleType.GetProperty("Name").GetValue(role, null);
+        }
+    }
+
+    public static string GetConfigurationSettingValue(string name)
+    {
+        if (!IsAvailable) throw new Exception("Role environment is not available, please check IsAvailable before calling this method!");
+
+        return (string)roleEnvironmentType.GetMethod("GetConfigurationSettingValue").Invoke(null, new object[] { name });
+    }
+
+    public static bool TryGetConfigurationSettingValue(string name, out string setting)
+    {
+        if (!IsAvailable) throw new Exception("Role environment is not available, please check IsAvailable before calling this method!");
+
+        setting = string.Empty;
+        bool result;
+        try
+        {
+            setting = (string)roleEnvironmentType.GetMethod("GetConfigurationSettingValue").Invoke(null, new object[] { name });
+            result = !string.IsNullOrEmpty(setting);
+        }
+        catch
```

why empty catch. can it be more specific